bolt-modules/log/lib/puppet/functions/log/warn.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+require 'bolt/util/format'
+
+# Log a warning message.
+#
+# Messages logged at this level typically include messages about deprecated
+# behavior or potentially harmful situations that might affect the plan run.
+# For example, you might log a message at the `warn` level if you are
+# planning to make a breaking change to your plan in a future release and
+# want to notify users.
+#
+# See [Logs](logs.md) for more information about Bolt's log levels.
+#
+# > **Note:** Not available in apply block
+Puppet::Functions.create_function(:'log::warn') do
+  # Log a warning message.
+  # @param message The message to log.
+  # @example Log a warning message
```

Comment:
```suggestion
  # @example Log a warn message
```
CC @hestonhoffman no idea which is right.

Reply:
I'd lean towards "warning" here. We should change the references to "warn logs" in logs.md too. These log entries are kinda weird cos they're all different parts of speech.